src/vs-bicep/Bicep.VSLanguageServerClient.IntegrationTests/Utilities/CompletionsUtility.cs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Linq;
+using Microsoft.Test.Apex.Editor;
+using Microsoft.Test.Apex.VisualStudio.Editor;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Bicep.VSLanguageServerClient.IntegrationTests.Utilities
+{
+    public static class CompletionsUtility
+    {
+        public static IVisualStudioCompletionListTestExtension? GetCompletionList(IVisualStudioTextEditorTestExtension editor, bool invokeIfNotPresent)
+        {
+            return GetCompletionList(editor, invokeIfNotPresent, ignoreExceptions: true);
+        }
+
+        public static void VerifyCompletions(IVisualStudioTextEditorTestExtension editor, string[] expectedCompletionTexts)
+        {
+            IVisualStudioCompletionListTestExtension? completionListTestExtension = GetCompletionList(editor, true);
+
+            if (completionListTestExtension is null)
+            {
+                Assert.IsNull(completionListTestExtension, "Completion list is null");
+                return;
+            }
+
+            CompletionList completionList = completionListTestExtension.Items;
+
+            foreach (var text in expectedCompletionTexts)
+            {
+                Assert.IsTrue(completionList.Any(e => e.Text == text));
+            }
+        }
+
+        private static IVisualStudioCompletionListTestExtension? GetCompletionList(IVisualStudioTextEditorTestExtension editor, bool invokeIfNotPresent, bool ignoreExceptions)
+        {
+            IVisualStudioCompletionListTestExtension? completionList = null;
+            try
+            {
+                completionList = editor.Intellisense.GetActiveCompletionList();
+                if (completionList == null)
+                {
+                    if (invokeIfNotPresent)
+                    {
+                        try
+                        {
+                            editor.Intellisense.ShowCompletionList();
+                        }
+                        catch
+                        {
+                            // This can throw when the list was coming up when we were invoking it
+                            if (!ignoreExceptions)
+                            {
+                                throw;
+                            }
+                        }
+
+                        completionList = WaitForExtensions.TryIsNotNull(() => editor.Intellisense.GetActiveCompletionList(), 2000);
+                    }
+                }
+            }
+            catch
+            {
```

Comment:
Consider adding a warning about the ignored exception into the log?

Reply:
Cleaned up to not ignore exceptions. The test I added doesn't involve any typing scenario. I'll revisit this if we ever encounter any issues around this utility/test.
https://github.com/Azure/bicep/commit/dc2779a89b6b66b4cc30b95c44c1035781cf59e5